src/connector/src/sink/iceberg/mod.rs

```diff
@@ -1315,14 +1416,75 @@ impl IcebergSinkCommitter {
 
 #[async_trait::async_trait]
 impl SinkCommitCoordinator for IcebergSinkCommitter {
-    async fn init(&mut self) -> Result<()> {
+    async fn init(&mut self) -> Result<Option<u64>> {
+        if self.is_exactly_once {
+            tracing::info!("Icberg sink coordinator initing");
+            if self
+                .iceberg_sink_has_pre_commit_metadata(&self.db, self.param.sink_id.sink_id())
+                .await?
+            {
+                let ordered_metadata_list_by_end_epoch = self
+                    .get_pre_commit_info_by_sink_id(&self.db, self.param.sink_id.sink_id())
+                    .await?;
+
+                let mut last_recommit_epoch = 0;
+                for (end_epoch, sealized_bytes, snapshot_id, committed) in
+                    ordered_metadata_list_by_end_epoch
+                {
+                    let write_results_bytes = deserialize_metadata(sealized_bytes);
+                    let mut write_results = vec![];
+
+                    for each in write_results_bytes {
+                        let write_result = IcebergCommitResult::try_from_sealized_bytes(each)?;
+                        write_results.push(write_result);
+                    }
+
+                    match (
+                        committed,
+                        self.is_snapshot_id_valid(&self.config, snapshot_id).await?,
+                    ) {
+                        (true, _) => {
+                            tracing::info!(
+                                "All data in log store has been written into external sink, skip"
+                            );
+                        }
+                        (false, true) => {
+                            // skip
+                            tracing::info!(
+                                "All pre-commit files have been successfully committed into iceberg and do not need to be committed again, mark it as committed"
+                            );
+                            self.mark_row_is_committed_by_sink_id_and_end_epoch(
+                                &self.db,
+                                self.sink_id,
+                                end_epoch,
+                            )
+                            .await?;
+                        }
+                        (false, false) => {
+                            tracing::info!(
+                                "There are files that were not successfully committed; re-commit these files."
+                            );
+                            self.re_commit(end_epoch, write_results).await?;
```

Comment:
If the snapshot_id is not contained in iceberg table, we should use this snapshot id to commit, however, we didn't use it here.